packages/arb-tx-aggregator/txaggregator/server.go

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2020, Offchain Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package txaggregator
+
+import (
+	"context"
+	"errors"
+	"log"
+	"math/big"
+	"sync"
+	"time"
+
+	errors2 "github.com/pkg/errors"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+
+	"github.com/offchainlabs/arbitrum/packages/arb-util/common"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/hashing"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/arbbridge"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/message"
+)
+
+const maxTransactions = 50
+
+type Server struct {
+	rollupAddress common.Address
+	globalInbox   arbbridge.GlobalInbox
+
+	sync.Mutex
+	valid        bool
+	transactions []message.Transaction
+	signatures   [][65]byte
+}
+
+// NewServer returns a new instance of the Server class
+func NewServer(ctx context.Context, globalInbox arbbridge.GlobalInbox, rollupAddress common.Address) *Server {
+	server := &Server{rollupAddress: rollupAddress, globalInbox: globalInbox, valid: true}
+
+	go func() {
+		ticker := time.NewTicker(common.NewTimeBlocksInt(5).Duration())
+		defer ticker.Stop()
+		for {
```

Comment:
Fixed

Reply:
I don't want to wait for a full batch since that would block unnecessarily. I changed it to only submit once per 5 seconds when the batches aren't full, but to keep submitting with no pause when they are full.